lib/dal/src/module.rs

```diff
@@ -402,4 +410,106 @@ impl Module {
 
         Ok(modules)
     }
+
+    /// Takes in a list of [`LatestModules`](si_frontend_types::LatestModule) and creates a
+    /// [`SyncedModules`](si_frontend_types::SyncedModules) object with them. The object enables callers to know what
+    /// [`Modules`](Module) can be upgraded and installed.
+    #[instrument(
+        name = "module.sync"
+        skip_all,
+        level = "debug",
+    )]
+    pub async fn sync(
+        ctx: &DalContext,
+        latest_modules: Vec<frontend_types::LatestModule>,
+    ) -> ModuleResult<frontend_types::SyncedModules> {
```

Comment:
This should be picked up now. Considered adding `latest_modules` to the fields, but the object could be pretty big since it is _all_ latest modules.
cc @stack72 @jobelenus